src/frontend/rustdoc/types.rs

```diff
@@ -0,0 +1,551 @@
+// Adopted from: rust-lang/rust, c254a15906d6ad64821659b50c4265c96b113159, src/librustdoc/json/types.rs 
```

Comment:
Is this not available as crate / library somehow e.g. through extern-crate magic?